docs/src/main/asciidoc/spanner.adoc

```diff
@@ -84,47 +325,74 @@ public class MyApplication {
+In the example above, the {spring-data-commons-ref}/#repositories.query-methods[query methods]
+in `TradeRepository` are generated based on the name of the methods, using the {spring-data-commons-ref}#repositories.query-methods.query-creation[Spring Data Query creation naming convention].
 
-|===
-| Name | Description | Optional | Default value
-| `spring.cloud.gcp.spanner.projectId` | the GCP project ID to use | Yes | provided by Spring Cloud GCP
-| `spring.cloud.gcp.spanner.instanceId` | Spanner instance to use | No |
-| `spring.cloud.gcp.spanner.database` |
-Spanner database to use | No |
-|===
+`List<Trade> findByAction(String action)` would translate to a `SELECT * FROM trades WHERE action = ?`.
+
+The function `List<Trade> findTop3DistinctByActionAndSymbolOrTraderIdOrderBySymbolDesc(String action, String symbol, String traderId);` will be translated as the equivalent of this SQL query:
+
+[source, sql]
+----
+SELECT DISTINCT * FROM trades
+WHERE ACTION = ? AND SYMBOL = ? AND Or TRADER_ID = ?
+ORDER BY SYMBOL DESC
+LIMIT 3
+----
+
+==== Mapping SQL to repository methods
+
+
+The example above for `List<Trade> fetchByActionNamedQuery(String action)` does not match the {spring-data-commons-ref}#repositories.query-methods.query-creation[Spring Data Query creation naming convention], we have to map a Spanner parametrized SQL query to it.
 
-Custom queries can be defined in properties files provided to the annotation
-`@EnableSpannerRepositories(namedQueriesLocation = "classpath:/[my-named-queries.properties file]")`.
-These query settings take the form `[ClassName].[QueryMethodName]=[query string with tags]`, where Spanner
-tags for parameters have the format `@[tagName]`. The actual names of the tags do not matter as long
-their order matches that defined in the repository interface.
+The SQL query for the method can be mapped to repository methods in one of two ways:
 
+ * `namedQueries` properties file
+ * using the `@Query` annotation
+
+In the example above the `namedQueriesLocation` attribute on `@EnableSpannerRepositories` is pointing to the `META-INF/spanner-named-queries.properties` file. You can specify the query for a method in the properties file by providing the SQL as the value for the "interface.method" property:
+
+[source, properties]
+----
+Trade.fetchByActionNamedQuery=SELECT * FROM trades WHERE trades.action = @tag0`
+----
+
+Using the `@Query` annotation:
+
+[source, java]
+----
+  @Query("SELECT * FROM trades WHERE trades.action = @tag0")
+  List<Trade> fetchByActionNamedQuery(String action);
+----
+
+Table names can be used directly, for example, "trades" in the above example. Alternatively, table names can be resolved from the `@Table` annotation on domain classes as well. In this case, the query should refer to table names with fully qualified classnames between `%` characters: `%fully.qualified.ClassName%`. A full example would look like this:
```

Comment:
This text mentions the % but we changed to using :  ( and the example is correctly using that)